src/Core/Renderer.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @file src/Core/Renderer.php
+ */
+
+namespace Friendica\Core;
+
+use Exception;
+use Friendica\BaseObject;
+use Friendica\Core\System;
+use Friendica\Render\FriendicaSmarty;
+
+/**
+ * @brief This class handles Renderer related functions.
+ */
+class Renderer extends BaseObject
+{
+    /**
+     * @brief This is our template processor
+     *
+     * @param string|FriendicaSmarty $s The string requiring macro substitution or an instance of FriendicaSmarty
+     * @param array $r                  key value pairs (search => replace)
+     * 
+     * @return string substituted string
+    */
+    public static function replaceMacros($s, $r)
+    {
+        $stamp1 = microtime(true);
+        $a = self::getApp();
+
+        // pass $baseurl to all templates
+        $r['$baseurl'] = System::baseUrl();
+        $t = $a->getTemplateEngine();
```

Comment:
Please move the template engine property to this class. You'll need an `init()` class at the spot in `App` where the templating engine is initialized.

Reply:
inside the `init()` you could take the `App` as a parameter and save it as
```php
private static $app;
```
so there is no need to have the `extend BaseObject` anymore.
Because it is called inside the `App` itself :-)

Reply:
The delim functions and props were easy enough, but I'm not fully following the template engine/init function request.

Reply:
You can just move `App->registerTemplateEngine`, `App->getTemplateEngine` and `App->setTemplateEngine` and their associated properties to `Core\Renderer`.